test/Microsoft.TemplateEngine.Mocks/MockTemplateInfo.cs

```diff
@@ -1,87 +1,342 @@
+            {
+                choicesDict.Add(choice, null);
+            };
+            return new CacheTag(tagDescription, choicesDict, defaultValue, defaultIfOptionWithoutValue);
+        }
+
+        #region XUnitSerializable implementation
+        public void Deserialize(IXunitSerializationInfo info)
+        {
+            Name = info.GetValue<string>("template_name");
+            Precedence = info.GetValue<int>("template_precedence");
+            Identity = info.GetValue<string>("template_identity");
+            GroupIdentity = info.GetValue<string>("template_group");
+            Description = info.GetValue<string>("template_description");
+            Author = info.GetValue<string>("template_author");
+            HasScriptRunningPostActions = info.GetValue<bool>("template_hasPostActions");
+
+            _tags = JsonConvert.DeserializeObject<Dictionary<string, string[]>>(info.GetValue<string>("template_tags"));
+            _cacheParameters = JsonConvert.DeserializeObject<string[]>(info.GetValue<string>("template_params"));
+            _baselineInfo = JsonConvert.DeserializeObject<string[]>(info.GetValue<string>("template_baseline"));
+            _classifications = JsonConvert.DeserializeObject<string[]>(info.GetValue<string>("template_classifications"));
+            _shortNameList = JsonConvert.DeserializeObject<string[]>(info.GetValue<string>("template_shortname"));
+        }
+
+        public void Serialize(IXunitSerializationInfo info)
+        {
+            info.AddValue("template_name", Name, typeof(string));
+            info.AddValue("template_shortname", JsonConvert.SerializeObject(_shortNameList), typeof(string));
+            info.AddValue("template_precedence", Precedence, typeof(int));
+            info.AddValue("template_identity", Identity, typeof(string));
+            info.AddValue("template_group", GroupIdentity, typeof(string));
+            info.AddValue("template_description", Description, typeof(string));
+            info.AddValue("template_author", Author, typeof(string));
+            info.AddValue("template_hasPostActions", HasScriptRunningPostActions, typeof(bool));
+
+            info.AddValue("template_tags", JsonConvert.SerializeObject(_tags), typeof(string));
+            info.AddValue("template_params", JsonConvert.SerializeObject(_cacheParameters), typeof(string));
+            info.AddValue("template_baseline", JsonConvert.SerializeObject(_baselineInfo), typeof(string));
+            info.AddValue("template_classifications", JsonConvert.SerializeObject(_classifications), typeof(string));
+        }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.Append("Short name:" + string.Join(",", _shortNameList) + ";");
+            sb.Append("Identity:" + Identity + ";");
+
+            if (string.IsNullOrEmpty(GroupIdentity))
+            {
+                sb.Append("Group:<not set>;");
```

Comment:
minor:
`StringBuilder::Append` returns `StringBuilder`.  Consider `_ = sb.Append("Group:<not set>;");`